lib/improver/tests/ensemble_calibration/ensemble_calibration/test_EstimateCoefficientsForEnsembleCalibration.py

```diff
@@ -580,67 +541,49 @@ def test_mean_predictor_estimate_coefficients_nans(self):
         """
-        data = np.array([1, 1, 2.666633, 1], dtype=np.float32)
-
-        current_forecast_predictor = self.cube.collapsed(
-            "realization", iris.analysis.MEAN)
-        current_forecast_predictor.data = (
-            current_forecast_predictor.data.filled())
-        truth = self.cube.collapsed("realization", iris.analysis.MAX)
-        distribution = "gaussian"
-        desired_units = "degreesC"
-        predictor_of_mean_flag = "mean"
+        data = np.array([0., 1., 1., 1.], dtype=np.float32)
         estimate_coefficients_from_linear_model_flag = True
 
-        current_forecast_predictor.data[0][0][0] = np.nan
+        self.current_forecast_predictor_mean.data = (
+            self.current_forecast_predictor_mean.data.filled())
+        self.current_forecast_predictor_mean.data[0][0] = np.nan
 
-        plugin = Plugin(distribution, desired_units)
+        plugin = Plugin(self.distribution, self.desired_units)
         result = plugin.compute_initial_guess(
-            truth, current_forecast_predictor, predictor_of_mean_flag,
+            self.truth, self.current_forecast_predictor_mean,
+            self.predictor_of_mean_flag,
             estimate_coefficients_from_linear_model_flag)
-
         self.assertArrayAlmostEqual(result, data)
 
 
-class Test_estimate_coefficients_for_ngr(IrisTest):
+class Test_estimate_coefficients_for_ngr(
+        SetupCubes, EnsembleCalibrationAssertions):
 
     """Test the estimate_coefficients_for_ngr plugin."""
 
     @ManageWarnings(
         ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
     def setUp(self):
         """Set up multiple cubes for testing."""
-        self.current_temperature_forecast_cube = (
-            add_forecast_reference_time_and_forecast_period(
-                set_up_temperature_cube()))
-
-        self.historic_temperature_forecast_cube = (
-            _create_historic_forecasts(self.current_temperature_forecast_cube))
-
-        self.temperature_truth_cube = (
-            _create_truth(self.current_temperature_forecast_cube))
-
-        self.current_wind_speed_forecast_cube = (
-            add_forecast_reference_time_and_forecast_period(
-                set_up_wind_speed_cube()))
-
-        self.historic_wind_speed_forecast_cube = (
-            _create_historic_forecasts(self.current_wind_speed_forecast_cube))
-
-        self.wind_speed_truth_cube = (
-            _create_truth(self.current_wind_speed_forecast_cube))
+        super().setUp()
+        self.current_cycle = "20171110T0000Z"
+        self.distribution = "gaussian"
 
         self.coeff_names = ["gamma", "delta", "alpha", "beta"]
+        self.coeff_names_realizations = (
+            ['gamma', 'delta', 'alpha', 'beta0', 'beta1', 'beta2'])
+
+        self.expected_mean_predictor_gaussian = (
+            [-0., 0.4888, 23.4351, 0.9129])
+        self.expected_mean_predictor_truncated_gaussian = (
+            [-0., 1.5434, -0.514, 0.94])
```

Comment:
This is just because the input to the Gaussian tests are temperatures in Kelvin, so the temperatures are around ~273.15K, whereas the wind speed is in m/s, so the wind speed values are slightly positive values. Given that these input values are quite different in size, it would be expected that these coefficients would differ.

Reply:
OK.  This looks better now anyway at the right level of precision, and it seems like the coefficient I would expect to be similar from your explanation (beta) is similar.  Thanks for adding the comments, they clarify things.